src/modules/Speech_Recognition/Whisper.py

```diff
@@ -63,6 +80,40 @@ def transcribe_with_whisper(
     keep_numbers: bool = False,
 ) -> TranscriptionResult:
     """Transcribe with whisper"""
+    # Info: Monkey Patch FasterWhisperPipeline.detect_language to include error handling for low confidence Test with https://www.youtube.com/watch?v=tcV7VN3l3bY
+    src = textwrap.dedent(inspect.getsource(whisperx.asr.FasterWhisperPipeline.detect_language))
+    # Replace the relevant part of the method
+    start_token = "if audio.shape[0] < N_SAMPLES:"
+    end_token = "return language"
+    replacement = """\
+    #Added imports
+    from modules.console_colors import ULTRASINGER_HEAD, blue_highlighted, red_highlighted
+    from Settings import Settings
+    #End Import addition
+    if audio.shape[0] < N_SAMPLES:
+        print("Warning: audio is shorter than 30s, language detection may be inaccurate.")
+    model_n_mels = self.model.feat_kwargs.get("feature_size")
+    segment = log_mel_spectrogram(audio[: N_SAMPLES],
+                                    n_mels=model_n_mels if model_n_mels is not None else 80,
+                                    padding=0 if audio.shape[0] >= N_SAMPLES else N_SAMPLES - audio.shape[0])
+    encoder_output = self.model.encode(segment)
+    results = self.model.model.detect_language(encoder_output)
+    language_token, language_probability = results[0][0]
+    language = language_token[2:-2]
+    print(f"Detected language: {language} ({language_probability:.2f}) in first 30s of audio...")
+    #Added handling for low detection probability (test with https://www.youtube.com/watch?v=tcV7VN3l3bY)
+    if language_probability < Settings.CONFIDENCE_THRESHOLD:
+        print(f"{ULTRASINGER_HEAD} {red_highlighted('Warning:')} Language detection probability for detected language {language} is below {Settings.CONFIDENCE_THRESHOLD}, results may be inaccurate.")
+        language_response = input(f"{ULTRASINGER_HEAD} Do you want to continue with {language} (default) or override with another language (y)? (y/n): ").strip().lower() == 'y'
```

Comment:
This would stop our automation.
If a Ui would use UltraSinger, this would cause a freeze.
Or if we have 100 songs in a folder that we want to run automatically overnight, this would also cause a freeze.

Maybe you should just warn them and say ‘Re-run with --language XY’?
Or wait x seconds for an input and select the default language?

Maybe it would also be a good idea to put a log.txt with warnings and settings in the output folder :D

Reply:
https://github.com/rakuri255/UltraSinger/issues/201

Reply:
So, I like the timeout idea and will implement it. As far as the freeze/automation question  goes, it is more complicated, especially since subsequent tasks also rely on the language code. Take a look at the fallback I put in for now. If no user response is given, the code will attempt to get language code from the user's locale, as I think this would be the safest assumption)

Reply:
Can you take another look to make sure we are good?